engines/base_parser_engine.py

```diff
@@ -315,9 +331,12 @@ def _parse_death(self, match, round):
         death_targets = [int(s) for s in text.split()]
         for t in death_targets:
             previous_round = self._get_previous_round(round)
-            death_method = self.night_deaths[previous_round][t]
-            self.clean_data[t]['death_round'] = previous_round
-            self.clean_data[t]['death_method'] = death_method
+            if t in self.night_deaths[previous_round]:
```

Comment:
If I understand this correctly, this is for magician operations. So I still think this should go to magician engine, not base engine. And we could use some comments here